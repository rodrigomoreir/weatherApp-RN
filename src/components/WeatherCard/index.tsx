import React from 'react';
import { Text } from 'react-native';

import {
  StyledContainer,
  StyledLeftContent,
  StyledRightContent,
  StyledCityContent,
  StyledTitle,
  StyledSubtitle,
  StyledTemperature
} from './styles';

interface Props {
  city: string;
  hour: string;
  weather: string;
  temperature: string;
  tempMaxAndMin: string;
}

export const WeatherCard = ({ city, hour, weather, temperature, tempMaxAndMin }: Props) => {
  return (
    <StyledContainer>
      <StyledLeftContent>
        <StyledCityContent>
          <StyledTitle>{city}</StyledTitle>
          <StyledSubtitle>{hour}</StyledSubtitle>
        </StyledCityContent>
        <StyledSubtitle>{weather}</StyledSubtitle>
      </StyledLeftContent>
      <StyledRightContent>
        <StyledTemperature>{temperature}</StyledTemperature>
        <StyledSubtitle>{tempMaxAndMin}</StyledSubtitle>
      </StyledRightContent>
    </StyledContainer>
  );
}
