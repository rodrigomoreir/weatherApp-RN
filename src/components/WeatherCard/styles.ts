import styled from 'styled-components/native';

export const StyledContainer = styled.TouchableOpacity`
  height: 120px;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.card};
  border-width: 0.5px;
  border-color: ${({ theme }) => theme.colors.border};
  margin-bottom: 10px;
  border-radius: 20px;
  padding: 15px;
  flex-direction: row;
  justify-content: space-between;
`
export const StyledLeftContent = styled.View`
  flex: 1;
  align-items: flex-start;
  justify-content: space-between;
`

export const StyledRightContent = styled.View`
  flex: 1;
  align-items: flex-end;
  justify-content: space-between;
`
export const StyledCityContent = styled.View`
`

export const StyledTitle = styled.Text`
  font-size: 20px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.textPrimary};
`

export const StyledSubtitle = styled.Text`
  font-size: 13px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.textSecondary};
`

export const StyledTemperature = styled.Text`
  font-size: 30px;
  color: ${({ theme }) => theme.colors.textPrimary};
`
