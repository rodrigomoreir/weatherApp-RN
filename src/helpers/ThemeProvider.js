import React from 'react';
import { ThemeProvider as DefaultThemeProvider } from 'styled-components/native';
import PropTypes from 'prop-types';

import { defaultTheme } from '../global/styles/theme';


const ThemeProvider = ({ children }) => (
  <DefaultThemeProvider theme={defaultTheme}>{children}</DefaultThemeProvider>
);

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
};

export default ThemeProvider;
