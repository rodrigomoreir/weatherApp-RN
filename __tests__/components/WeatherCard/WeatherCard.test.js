import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import WeatherCard from '../../../src/components/WeatherCard';

import { renderWithTheme, matchSnapshotWithTheme } from '../../../src/helpers/testing'

describe('Card', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(
      <WeatherCard
        city='City'
        hour='hour'
        weather='weather'
        temperature='temperature'
        tempMaxAndMin='tempMaxAndMin'
        onPress={jest.fn()}
      />
    );
  });

  test('renders correctly', () => {
    renderWithTheme(
      <WeatherCard
        city='City'
        hour='hour'
        weather='weather'
        temperature='temperature'
        tempMaxAndMin='tempMaxAndMin'
        onPress={jest.fn()}
      />
    );
  });

  test('callback is called on press', () => {
    const onPressMock = jest.fn();
    const { getByText } = renderWithTheme(
      <WeatherCard
        city='City'
        hour='hour'
        weather='weather'
        temperature='temperature'
        tempMaxAndMin='tempMaxAndMin'
        onPress={onPressMock}
      />
    );
    fireEvent.press(getByText('City'));
    expect(onPressMock).toHaveBeenCalled();
  });
});
