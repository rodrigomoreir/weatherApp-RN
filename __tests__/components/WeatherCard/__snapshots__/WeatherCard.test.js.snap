// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<View
  animate={
    Object {
      "opacity": 1,
      "translateY": 0,
    }
  }
  from={
    Object {
      "opacity": 0,
      "translateY": 50,
    }
  }
  style={
    Array [
      Object {
        "opacity": [MockFunction],
        "transform": [MockFunction],
      },
    ]
  }
  transition={
    Object {
      "duration": 700,
      "type": "timing",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F5F8FA",
        "borderColor": "#808080",
        "borderRadius": 20,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          City
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#969CB2",
                "fontSize": 13,
                "fontWeight": "bold",
              },
            ]
          }
        >
          hour
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#969CB2",
              "fontSize": 13,
              "fontWeight": "bold",
            },
          ]
        }
      >
        weather
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 30,
            },
          ]
        }
      >
        temperature
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#969CB2",
              "fontSize": 13,
              "fontWeight": "bold",
            },
          ]
        }
      >
        tempMaxAndMin
      </Text>
    </View>
  </View>
</View>
`;
